zetaclient/zetacore_observer.go

```diff
@@ -227,6 +234,40 @@ func (co *CoreObserver) startSendScheduler() {
 	}
 }
 
+// Bitcoin outtx is processed in a different way
+// 1. schedule one keysign on each ticker
+// 2. schedule keysign only when nonce-mark UTXO is available
+// 3. stop processing when pendingNonce/lookahead is reached
+//
+// Returns whether to stop processing
+func (co *CoreObserver) processBitcoinOutTx(outTxMan *OutTxProcessorManager, idx int, send *types.CrossChainTx, signer ChainSigner, ob ChainClient, currentHeight uint64) bool {
+	params := send.GetCurrentOutTxParam()
+	nonce := params.OutboundTxTssNonce
+	lookahead := uint64(ob.GetCoreParams().OutboundTxScheduleLookahead)
+	outTxID := fmt.Sprintf("%s-%d-%d", send.Index, params.ReceiverChainId, nonce)
+
+	// start go routine to process outtx
+	outTxMan.StartTryProcess(outTxID)
+	co.logger.ZetaChainWatcher.Debug().Msgf("Sign bitcoin outtx %s with value %d\n", outTxID, params.Amount)
+	go signer.TryProcessOutTx(send, outTxMan, outTxID, ob, co.bridge, currentHeight)
```

Comment:
It's a bit hard to visualize how concurrency is managed with this Go routine. Is the Go routine communicating with `outTxMan`? What is the process to wait/determine the out tx is processed?

Reply:
The logic here is to make sure there will be only one go routine processing the cctx with identifier `outTxID`. It's controlled by a defer function inside function `TryProcessOutTx` (both EVM and Bitcoin chains)
https://github.com/zeta-chain/node/blob/btc-keysign-schedule/zetaclient/btc_signer.go#L205